components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -1467,4 +1467,40 @@ public static String getAuthorizationHeaderFromSwagger(Map<String, Object> exten
         return authorizationHeader == null ? null : authorizationHeader.toString();
     }
 
+    /**
+     * This method returns extension of application security types related to micro-gw
+     *
+     * @param extensions Map<String, Object>
+     * @return String
+     * @throws APIManagementException throws if an error occurred
+     */
+    public static List<String> getApplicationSecurityTypes(Map<String, Object> extensions) throws APIManagementException {
+        ObjectMapper mapper = new ObjectMapper();
+        List<String> appSecurityTypes = new ArrayList<>();
+        if (extensions.containsKey(APIConstants.X_WSO2_APP_SECURITY)) {
+            Object applicationSecurityTypes = extensions.get(APIConstants.X_WSO2_APP_SECURITY);
+            ObjectNode appSecurityTypesNode = mapper.convertValue(applicationSecurityTypes, ObjectNode.class);
+            appSecurityTypes = mapper.convertValue(appSecurityTypesNode.get("security-types"), ArrayList.class);
+        }
+        return appSecurityTypes;
+    }
+
+    /**
+     * This method returns extension of application security types state related to micro-gw
+     *
+     * @param extensions Map<String, Object>
+     * @return boolean
```

Comment:
here is no use of this doc comment if you don't add a description on what it returns.

Reply:
Fixed with c1f846270025eb67fcd8e92c102a8539e1e699a3